components/data-apps/index.md

```diff
@@ -0,0 +1,116 @@
+This type of application is usually custom tailored to tackle a specific problem and entails a dynamic, purpose-built user experience.
+
+Some examples of data apps could be recommendation engines, interactive segmentation, AI integration, data visualization, 
+customized internal reporting tools for business teams, financial app to get insights on your spend patterns etc.
+
+Data Apps could be written in any language, but for now Keboola only supports apps written in [Streamlit](https://streamlit.io/), 
+which is Python framework for rapid development of such applications.
+
+As mentioned above, Data App is a simple web application, which can be deployed inside a Keboola Project 
+and can publicly accessible also from outside the project. 
+Meaning that the users accessing your Data App don't need access to a Keboola project.
+
+## Create a Data App
+There are two ways how to create a Data App in Keboola. Choose deployment type which suits your needs:
+- **Code** - Just paste a Streamlit code to create a simple Data App. 
+- **Git repository** - Specify git repository with Streamlit app sources. More suitable for complex applications.
+
+{: .image-popup}
+![Code - main menu](/components/data-apps/data apps - main menu.png)
+
+### Code
+For a simple use-cases, where your Streamlit code fits into a one page, paste the code directly into a textarea. 
+Ideal for very simple apps or testing things. Try it immediately with [this example from Streamlit docs](- https://docs.streamlit.io/library/get-started/create-an-app#lets-put-it-all-together).
+
+{: .image-popup}
+![Code - code](/components/data-apps/data apps - hello world - code.png)
+
+#### Packages
+To use additional packages, which are not already in our [Streamlit Base Image](#Base Image), enter it into the Packages field.
+
+{: .image-popup}
+![Packages](/components/data-apps/data apps - packages.png)
+
+### Git repository
+{% include warning.html content="In BETA we only support Github repositories." %}
+
+Please provide feedback using the feedback button in your project." %}
+If you have a complex application, you can push your app sources into Github and link the repository in this section.
+Provide Project URL, choose the right branch and finally select your main entrypoint file.
+
+{: .image-popup}
+![Git repository](/components/data-apps/data apps - git repository.png)
+
+## Secrets
+To provide your app with environment variables or sensitive information like credentials, api keys etc, enter them as key value pairs in the Secrets section.
+These secrets will be injected into the secrets.toml file upon deployment of the app. 
+[Read more about the Streamlit secrets](https://docs.streamlit.io/streamlit-community-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management).
+
+{: .image-popup}
+![Secrets](/components/data-apps/data apps - secrets.png)
+
+## Data loading
```

Comment:
This is data loading can we add a section about getting data back to keboola, please?